indexer/lib/advertisement-walker.js

```diff
@@ -169,52 +170,68 @@ export async function processNextAdvertisement ({
       state.status = `Walking the advertisements from ${state.head}, next step: ${state.tail}`
     }
 
-    const indexEntry = entry.pieceCid ? entry : undefined
+    if (error === 'CANNOT_FETCH_ENTRIES') {
+      state.entriesNotRetrievable = (state.entriesNotRetrievable ?? 0) + 1
+    } else if (error === 'CANNOT_DETERMINE_PIECE_CID') {
+      state.adsMissingPieceCID = (state.adsMissingPieceCID ?? 0) + 1
+    }
+
+    const indexEntry = entry?.pieceCid ? entry : undefined
     const finished = !state.tail
     return {
       newState: state,
       indexEntry,
       finished
     }
   } catch (err) {
-    let reason
-    if (err instanceof Error) {
-      const url = 'url' in err ? err.url : undefined
-      if ('serverMessage' in err && err.serverMessage) {
-        reason = err.serverMessage
-      } else if ('statusCode' in err && err.statusCode) {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} failed: ${err.statusCode}`
-      } else if (err.name === 'TimeoutError') {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} timed out`
-      } else if (
-        err.name === 'TypeError' &&
-        err.message === 'fetch failed' &&
-        err.cause &&
-        err.cause instanceof Error
-      ) {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} failed: ${err.cause.message}`
-      }
-    }
+    const errorDescription = describeFetchError(err, providerInfo.providerAddress)
 
     debug(
       'Cannot process provider %s (%s) advertisement %s: %s',
       providerId,
       providerInfo.providerAddress,
       state.tail,
-      reason ?? err
+      errorDescription ?? err
     )
-    state.status = `Error processing ${state.tail}: ${reason ?? 'internal error'}`
+    state.status = `Error processing ${state.tail}: ${errorDescription ?? 'internal error'}`
     return {
       newState: state,
       failed: true
     }
   }
 }
 
-/** @typedef {{
-    pieceCid: string | undefined;
-    payloadCid: string;
-}} AdvertisedPayload */
+/**
+ * @param {unknown} err
+ * @param {string} providerAddress
```

Comment:
```suggestion
 * @param {string} providerAddress
 * @returns {string}
```

Reply:
TypeScript infers the return type.

<img width="740" alt="Screenshot 2024-10-02 at 13 34 08" src="https://github.com/user-attachments/assets/3343285e-6b40-45ad-9739-eef7f3dc4020">

But I can see how it can be helpful to see the return type in the JSDoc comment. I'll add `@returns {string | undefined}`